File: UI/epreschool_mobile/lib/models/registrationModel.dart
import 'package:intl/intl.dart';

class RegistrationModel {
  late int id;
  late String firstName;
  late String lastName;
  late String userName;
  late String createdAt;
  late String phoneNumber;
  late int gender;
  late String email;
  late DateTime birthDate;
  late int kindergartenId;
  late String address;
  late String postCode;
  late int qualification;
  RegistrationModel();

  RegistrationModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    firstName = json['firstName'];
    lastName = json['lastName'];
    gender = json['gender'];
    // birthDate =
    //     DateFormat('dd.MM.yyyy').format(DateTime.parse(json['birthDate']));
    address = json['address'];
    postCode = json['postCode'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['firstName'] = firstName;
    data['lastName'] = lastName;
    data['gender'] = gender;
    data['userName'] = userName;
    data['email'] = email;
    data['phoneNumber'] = phoneNumber;
    data['birthDate'] = DateFormat('yyyy-MM-ddTHH:mm:ss').format(birthDate);
    data['address'] = address;
    data['postCode'] = postCode;
    data['kindergartenId'] = kindergartenId;
    data['qualification'] = qualification;
    data['isEmployed'] = false;
    data['employerName'] = "";
    data['employerPhoneNumber'] = "";
    data['employerAddress'] = "";
    return data;
  }
}

File: ePreschool.Core/Models/Registration/RegistrationModel.cs
using ePreschool.Core.Enumerations;
using ePreschool.Core.Models;

namespace ePreschool.Core
{
    public class RegistrationModel : BaseModel
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string UserName { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public Gender Gender { get; set; }
        public DateTime BirthDate { get; set; }
        public int KindergartenId { get; set; }
        public string Address { get; set; }
        public string PostCode { get; set; }
        public Qualification Qualification { get; set; }
        public bool IsEmployed { get; set; }
        public string? EmployerName { get; set; }
        public string? EmployerAddress { get; set; }
        public string? EmployerPhoneNumber { get; set; }
    }
}
